content/source/docs/enterprise/workspaces/variables.html.md

```diff
@@ -0,0 +1,94 @@
+
+```hcl
+{
+    us-east-1 = "image-1234"
+    us-west-2 = "image-4567"
+}
+```
+
+HCL can be used for Terraform variables, but not for environment variables. The HCL code you enter for values is interpreted by the same Terraform version that performs runs in the workspace. (See [How TFE Uses Variables](#how-tfe-uses-variables) below.)
+
+### Sensitive Values
+
+Terraform often needs cloud provider credentials and other sensitive information that shouldn't be widely available within your organization.
+
+To protect these secrets, you can mark any any Terraform or environment variable as sensitive data by clicking its "Sensitive" checkbox in edit mode.
+
+Marking a variable as sensitive severely limits how users can interact with it:
+
+- Nobody (including you) can view the value in TFE's UI or API. (However, Terraform runs will receive the value, and might print the value in logs if the configuration pipes the value through to an output or a resource parameter. Take care when writing your configurations.)
+- Nobody can edit the variable's name or value, change whether or not it's an HCL value, or mark it as non-sensitive. To alter anything about a sensitive variable, you must delete it and create a new variable with the same name.
+
+### Determining Variable Names
+
+Terraform Enterprise can't automatically discover variable names from a workspace's Terraform code. You must discover the necessary variable names by reading code or documentation, then enter them manually.
+
+If a required input variable is missing, Terraform plans in the workspace will fail and print an explanation in the log.
+
+## How TFE Uses Variables
+
+### Terraform Variables
+
+After retrieving the workspace's configuration, TFE uses a workspace's Terraform variables to write a `terraform.tfvars` file into the working directory. Terraform loads that `tfvars` file as part of the run.
+
+This means:
+
+- HCL variable values behave predictably, without needing to accommodate shell quoting.
+- You cannot commit a file named `terraform.tfvars` to version control, since TFE will overwrite it. (Note that you shouldn't check in `terraform.tfvars` even when running Terraform solely on the command line.)
+- In the event that HCL syntax changes slightly between Terraform versions, all values in a given workspace will be handled consistently, since the variables file is interpreted by the same Terraform version that loads the rest of the code.
+
+### Environment Variables
+
+TFE performs Terraform runs on disposable Linux worker VMs using a POSIX-compatible shell. Before running Terraform, TFE populates the shell with environment variables using the `export` command.
+
+### Secure Storage of Variables
+
+TFE's backend storage system encrypts variables on disk using AES-128, and uses Vault to protect the encryption key.
```

Comment:
Was this written assuming that Archivist was used for variable storage? It would probably be better to write this like:

> TFE encrypts all variable values securely using Vault's transit backend prior to saving them. This ensures that no out-of-band party is able to read these values without proper authorization.

Reply:
Yeah, the legacy TFE docs said "variables are stored in vault," and after looking into it for a while I landed on the Archivist README. 

I just read up briefly on the transit backend, but it's new to me so I want to make sure I understand the implications: This means the data is encrypted in its at-rest storage, right? Archivist doesn't necessarily understand how to decrypt it, but that's fine because access goes through Vault, so Vault handles the decryption on access? ...Did I get that right?

Reply:
Ryan's suggested statement is exactly what I'd like to read. It calls out the use of Vault transit backend and why it's used.

I don't think we need to call out how the unseal keys are managed or where the data gets stored, as that level of detail is only necessary for PTFE (not SaaS) and it'll be configured in the install docs for PTFE. (Though I'm guessing you want to know this for your own benefit, not necessarily for the docs.)